apps/vaults/components/details/actions/QuickActionsFrom.tsx

```diff
@@ -0,0 +1,93 @@
+import React from 'react';
+import {useActionFlow} from '@vaults/contexts/useActionFlow';
+import {useWeb3} from '@yearn-finance/web-lib/contexts/useWeb3';
+import {toAddress} from '@yearn-finance/web-lib/utils/address';
+import {formatAmount} from '@yearn-finance/web-lib/utils/format.number';
+import {formatCounterValue} from '@yearn-finance/web-lib/utils/format.value';
+import {Dropdown} from '@common/components/TokenDropdown';
+import {useWallet} from '@common/contexts/useWallet';
+import {useBalance} from '@common/hooks/useBalance';
+import {useTokenPrice} from '@common/hooks/useTokenPrice';
+import {handleInputChange} from '@common/utils';
+
+import type {ChangeEvent, ReactElement} from 'react';
+import type {TDropdownOption} from '@common/types/types';
+
+function	VaultDetailsQuickActionsFrom(): ReactElement {
+	const {isActive} = useWeb3();
+	const {balances} = useWallet();
+	const {
+		possibleOptionsFrom, selectedOptionFrom, onUpdateSelectedOptionFrom,
+		amount, onChangeAmount,
+		maxDepositPossible, isDepositing
+	} = useActionFlow();
+
+	const selectedFromBalance = useBalance(toAddress(selectedOptionFrom?.value));
+	const selectedOptionFromPricePerToken = useTokenPrice(toAddress(selectedOptionFrom?.value));
+
+	return (
+		<section aria-label={'FROM'} className={'flex w-full flex-col space-x-0 md:flex-row md:space-x-4'}>
+			<div className={'relative z-10 w-full space-y-2'}>
+				<div className={'flex flex-row items-baseline justify-between'}>
+					<label className={'text-base text-neutral-600'}>
+						{isDepositing ? 'From wallet' : 'From vault'}
+					</label>
+					<legend className={'font-number inline text-xs text-neutral-600 md:hidden'} suppressHydrationWarning>
+						{`You have ${formatAmount(selectedFromBalance.normalized)} ${selectedOptionFrom?.symbol || 'tokens'}`}
+					</legend>
+				</div>
+				{possibleOptionsFrom.length > 1 ? (
+					<Dropdown
+						defaultOption={possibleOptionsFrom[0]}
+						options={possibleOptionsFrom}
+						selected={selectedOptionFrom}
+						onSelect={(option: TDropdownOption): void => onUpdateSelectedOptionFrom(option)} />
```

Comment:
We need any opportunity we can get to reduce the verbosity of the code to make it more readable :)
```suggestion
						onSelect={onUpdateSelectedOptionFrom} />
```